Assets/Scripts/Gameplay/ConnectionManagement/ConnectionState/ConnectionState.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Unity.Multiplayer.Samples.BossRoom.Shared.Infrastructure;
+using Unity.Netcode;
+using UnityEngine;
+
+namespace Unity.Multiplayer.Samples.BossRoom
+{
+    /// <summary>
+    /// Base class representing a connection state. Contains a static member for each possible state.
+    /// </summary>
+    public abstract class ConnectionState
+    {
+        public static readonly OfflineConnectionState Offline = new OfflineConnectionState();
```

Comment:
This is interesting and I like this. I'd still like to have a quick convo on the other option we discussed, just to be thorough about this design decision.